service/ziti-tunnel/service/ipc.go

```diff
@@ -1100,10 +1128,82 @@ func handleEvents(isInitialized chan struct{}) {
 		case <-every5s.C:
 			s := rts.ToMetrics()
 
+			// broadcast metrics
 			rts.BroadcastEvent(dto.MetricsEvent{
 				StatusEvent: dto.StatusEvent{Op: "metrics"},
 				Identities:  s.Identities,
 			})
+
+			// refresh timeout of services
+			for _, id := range rts.ids {
+				zid := id.CId
+				if zid == nil || !zid.RefreshCheck(zid.MinTimeout, zid.MaxTimeout) {
+					continue
+				}
+
+				// refresh timeout
+				zid.Services.Range(func(key interface{}, value interface{}) bool {
+					//string, ZService
+					val := value.(*cziti.ZService)
+
+					if val.Service.Timeout <= 0 {
+						return true
+					}
+
+					var svcTimeout int32 = -1
+					for _, pc := range val.Service.PostureChecks {
+						if svcTimeout == -1 || svcTimeout > int32(pc.Timeout) {
+							svcTimeout = int32(pc.Timeout)
+						}
+						break
+					}
+					if (svcTimeout - int32(time.Since(zid.LastUpdatedTime).Seconds())) < 0 {
+						atomic.StoreInt32(&val.Service.Timeout, 0)
+					} else {
+						atomic.StoreInt32(&val.Service.Timeout, svcTimeout-int32(time.Since(zid.LastUpdatedTime).Seconds()))
+					}
+					return true
+				})
+
+			}
+
+		// notification message
+		case <-notificationFrequency.C:
+			cleanNotifications := make([]cziti.NotificationMessage, 0)
+			for _, id := range rts.ids {
+				if id.CId == nil || !id.CId.RefreshCheck(id.CId.MinTimeout, id.CId.MaxTimeout) {
+					continue
+				}
+				notificationMessage := ""
+
+				if (id.CId.MaxTimeout - int32(time.Since(id.CId.LastUpdatedTime).Seconds())) < 0 {
+					notificationMessage = fmt.Sprintf("All of the services of identity %s are timed out", id.Name)
+				} else if (id.CId.MinTimeout - int32(time.Since(id.CId.LastUpdatedTime).Seconds())) < 0 {
+					notificationMessage = fmt.Sprintf("Some of the services of identity %s are timed out", id.Name)
+				} else if (id.CId.MinTimeout - int32(time.Since(id.CId.LastUpdatedTime).Seconds())) < int32(time.Duration(20*time.Minute).Seconds()) {
```

Comment:
i believe `time.Duration` is not needed here - just the parens

Reply:
time Duration is in the notification object